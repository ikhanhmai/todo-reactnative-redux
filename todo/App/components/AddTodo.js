import React from 'react';
import {View, TextInput, Button} from 'react-native';
import {addTodo} from '../actions/index';
class AddTodo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  onChangeTextHandler = text => {
    this.setState({
      text: text,
    });
  };

  addTodoHandler = () => {
    this.props.dispatch(addTodo(this.state.text));
  };

  render() {
    return (
      <View>
        <TextInput
          placeholder="type to add todo"
          onChangeText={this.onChangeTextHandler}
        />
        <Button title={'Add Todo'} onPress={this.addTodoHandler} />
      </View>
    );
  }
}
export default AddTodo;
