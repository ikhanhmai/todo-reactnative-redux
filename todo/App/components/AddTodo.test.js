import React from 'react';
import {shallow} from 'enzyme';
import AddTodo from './AddTodo';

describe('AddTodo', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<AddTodo/>)
            expect(component).toMatchSnapshot()
        });
    });
});