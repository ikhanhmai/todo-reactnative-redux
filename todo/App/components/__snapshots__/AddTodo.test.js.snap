// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo Rendering should match to snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="type to add todo"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <Button
    onPress={[Function]}
    title="Add Todo"
  />
</View>
`;
